trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F

```diff
@@ -0,0 +1,350 @@
+    use module_read_timeslice_data
+    implicit none
+
+    ! Extend/derive hybrid struct from the abstract base
+    ! struct for reservoirs.
+    type, extends(reservoir_base_struct) :: persistence_levelpool_hybrid_struct
+
+        ! Define pointers to sub-structs / sub-objects to and
+        ! held by a level pool reservoir object.
+        type (hybrid_parameters_interface), pointer :: parameters => null()
+        type (hybrid_state_interface), pointer :: state => null()
+
+        logical :: pointer_allocation_guard = .false.
+
+    contains
+
+        procedure :: init => hybrid_struct_init
+        procedure :: destroy => hybrid_struct_destroy
+        procedure :: run_release => run_hybrid_release
+
+    end type persistence_levelpool_hybrid_struct
+
+contains
+
+    ! Hybrid Constructor
+    subroutine hybrid_struct_init(this, water_elevation,  &
+        lake_area, weir_elevation, weir_coeffecient, &
+        weir_length, orifice_elevation, orifice_coefficient, &
+        orifice_area, lake_max_water_elevation, initial_fractional_depth, &
+        lake_number, persistence_parameter_file, start_date, &
+        timeslice_path, observation_lookback_hours, &
+        observation_update_time_interval_seconds)
+        implicit none
+        class(persistence_levelpool_hybrid_struct), intent(inout) :: this ! object being initialized
+        real,    intent(inout) :: water_elevation           ! meters AMSL
+        real,    intent(in)    :: lake_area      		    ! area of lake (km^2)
+        real,    intent(in)    :: weir_elevation            ! bottom of weir elevation (meters AMSL)
+        real,    intent(in)    :: weir_coeffecient          ! weir coefficient
+        real,    intent(in)    :: weir_length               ! weir length (meters)
+        real,    intent(in)    :: orifice_elevation         ! orifice elevation (meters AMSL)
+        real,    intent(in)    :: orifice_coefficient       ! orifice coefficient
+        real,    intent(in)    :: orifice_area              ! orifice area (meters^2)
+        real,    intent(in)    :: lake_max_water_elevation  ! max water elevation (meters)
+        real,    intent(in)    :: initial_fractional_depth  ! initial fraction water depth
+        integer, intent(in)    :: lake_number               ! lake number
+        character(len=*),   intent(in) :: persistence_parameter_file
+        character(len=19),  intent(in) :: start_date
+        character(len=256), intent(in) :: timeslice_path
+        integer,            intent(in) :: observation_lookback_hours
+        integer,            intent(in) :: observation_update_time_interval_seconds
+        character(len=15)              :: lake_number_string
+
+        ! Create diagnostic log file
+        ! Uncomment this call only for development/debugging purposes
+        !call create_hybrid_diagnostic_log_file(lake_number)
+
+        ! Call to initialize read timeslice data object
+        call read_timeslice_data%init(start_date, &
```

Comment:
What is the scope of read_timeslice_data?  I think it is module_read_timeslice_data? I'm very confused that i'm not seeing `type(read_timeslice_data) :: read_reservoir_timeslice` nor `read_res_timeslice = read_timeslice_data` =  in this module.
... now i see line 61.... 
https://github.com/NCAR/wrf_hydro_nwm_public/pull/259/files?file-filters%5B%5D=.F&file-filters%5B%5D=.inc&file-filters%5B%5D=.json&file-filters%5B%5D=.namelist&file-filters%5B%5D=.nc&file-filters%5B%5D=.ncdf&file-filters%5B%5D=No+extension&file-filters%5B%5D=dotfile#diff-0966770ea36093ff0b4aa63fc5b9e738R61

why? why not composition. it's like half module programming, half OO.

Reply:
This is meant to be a singleton in this case.  One object shared across all reservoirs.  Each reservoir inits the object, but if it has already been initialized, nothing happens.  This also works in the distributed domain since each processes that contains a reservoir will initialize at some point, so each node gets a singleton timeslice reader.

Reply:
OK. That's clever. It's not clear. I think it deserves some commenting like you just provided. 
I wonder if there's a better way to express in code so it's clearer....

Reply:
I'm mostly wondering for my own curiosity. I think commenting it would be sufficient.

Reply:
@jdmattern-noaa fixing. Adding singleton comments.